pos_attachment/static/src/app/screens/product_screen/attachments_button.esm.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module */
+/*
+    Copyright 2024 Dixmit
+    License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+*/
+import {AttachmentPopup} from "../../utils/attachment_popup/attachment_popup.esm";
+import {Component} from "@odoo/owl";
+import {ConnectionLostError} from "@web/core/network/rpc_service";
+import {ProductScreen} from "@point_of_sale/app/screens/product_screen/product_screen";
+import {usePos} from "@point_of_sale/app/store/pos_hook";
+import {useService} from "@web/core/utils/hooks";
+
+export class AttachmentsButton extends Component {
+    setup() {
+        this.pos = usePos();
+        this.popup = useService("popup");
+    }
+
+    get attachments() {
+        const order = this.pos.get_order();
+        return order ? order.message_attachment_count : 0;
+    }
+
+    async click() {
+        const order = this.pos.get_order();
+        order.save_to_db();
+        this.env.services.ui.block();
+        const syncOrderResult = await this.pos.push_single_order(order);
+        this.env.services.ui.unblock();
+        if (syncOrderResult instanceof ConnectionLostError) {
+            return;
+        } else if (!syncOrderResult) {
+            return;
+        }
+        await this.popup.add(AttachmentPopup, {order: order});
+    }
+}
```

Comment:
Because I want to avoid to store the attachments inside the Database. It is certainly an issue, maybe it has more sense to add this at already validated orders. WDYT?